core/admin/mailu/limiter.py

```diff
@@ -68,9 +68,13 @@ def should_rate_limit_user(self, username, ip, device_cookie=None, device_cookie
             app.logger.warn(f'Authentication attempt from {ip} for {username} has been rate-limited.')
         return is_rate_limited
 
-    def rate_limit_user(self, username, ip, device_cookie=None, device_cookie_name=None):
+    def rate_limit_user(self, username, ip, device_cookie=None, device_cookie_name=None, password=''):
         limiter = self.get_limiter(app.config["AUTH_RATELIMIT_USER"], 'auth-user')
         if self.is_subject_to_rate_limits(ip):
+            truncated_password = hmac.new(bytearray(username, 'utf-8'), bytearray(password, 'utf-8'), 'sha256').hexdigest()[-6:]
+            if password and self.storage.get(f'dedup2-{username}-{truncated_password}') > 0:
```

Comment:
```suggestion
            if password and (self.storage.get(f'dedup2-{username}-{truncated_password}') > 0):
```

Make it explicitly clear that `and` takes precedence over `>`>